File: leetcode/2512.奖励最顶尖的 K 名学生.py
class Solution:
    def topStudents(self, positive_feedback: List[str], negative_feedback: List[str], report: List[str], student_id: List[int], k: int) -> List[int]:
        positive_feedback = Counter(positive_feedback)
        negative_feedback = Counter(negative_feedback)
        n = len(student_id)
        ans = []
        for i in range(n):
            t = 0
            for ss in report[i].split():
                if ss in positive_feedback:
                    t += 3
                elif ss in negative_feedback:
                    t -= 1
            ans.append([-t, student_id[i]])
        ans.sort()
        return [i for v, i in ans[:k]]
                

File: leetcode/2512.奖励最顶尖的 K 名学生.go
func topStudents(positive_feedback []string, negative_feedback []string, report []string, student_id []int, k int) []int {
    words := map[string]int{}
    for _, w := range positive_feedback {
        words[w] = 3
    }
    for _, w := range negative_feedback {
        words[w] = -1
    }
    type pair struct{
        sorce int
        id int
    }
    A := make([]pair, len(report))
    for i, r := range report {
        sorce := 0
        for _, w := range strings.Split(r, " ") {
            sorce += words[w]
        }
        A[i] = pair{sorce, student_id[i]}
    }
    sort.Slice(A, func(i, j int) bool {
        a, b := A[i], A[j]
        return a.sorce > b.sorce || a.sorce == b.sorce && a.id < b.id
    })
    res := make([]int, k)
    for i, p := range A[:k] {
        res[i] = p.id
    }
    return res
}